src/main/fc/core.c

```diff
@@ -986,7 +986,7 @@ void processRxModes(timeUs_t currentTimeUs)
     }
 
 #ifdef USE_ALTHOLD_MODE
-    if (ARMING_FLAG(ARMED) && IS_RC_MODE_ACTIVE(BOXALTHOLD) && (sensors(SENSOR_ACC)) && !failsafeIsActive()) {
+    if (ARMING_FLAG(ARMED) && IS_RC_MODE_ACTIVE(BOXALTHOLD) && (sensors(SENSOR_ACC)) && !failsafeIsActive() && isAirmodeActivated()) {
```

Comment:
Does this make 'air mode' mandatory for altitude maintenance?

Reply:
If we don't block alt hold until we are in the air, then it can be initiated while idling on the ground (by mistake).
Then the quad immediately gets hover throttle, and with Baro the air is compressed under the quad, and it seems like the quad just dropped, so it jumps up.  Even if the user keeps throttle at zero, it will take time for the setpoint to be low enough that the PIDs stop it bouncing.
Hence we need a way to stop it until we are in the air.
This was my first thought.
Another way is to not allow it to start unless throttle is up some way from idle.
But then we may have people panicking due to disorientation or loss of video or something, and dropping throttle to zero and simultaneously requesting position hold.
Ideally we would have some throttle delay test that detects that the quad is in the air.